src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = ({ ...state });
+  const renovatedState = [];
+
+  for (const action of actions) {
+    const extraDatavalue = action.extraData || {};
+    const typeOfAction = action.type;
+    const removeValue = action.keysToRemove || {};
+
+    switch (typeOfAction) {
+      case 'addProperties':
+        for (const newProperty in extraDatavalue) {
+          newState[newProperty] = extraDatavalue[newProperty];
+        }
+
+        break;
+
+      case 'removeProperties':
+        for (let j = 0; j < removeValue.length; j++) {
+          delete newState[removeValue[j]];
+        }
+
+        break;
+
+      default:
+        Object.keys(newState).forEach(key => delete newState[key]);
+    }
```

Comment:
if I understand correct this code **"default: Object.keys(newState).forEach(key => delete newState[key])"** are clearing the properties of the newState object. The **default** case should not be used to clear the state object. You should use: default:throw new Error(``). But I'm not 100% sure about it, so I confirm the completion of the task, and the mentor will confirm or discard my opinion.